### PR TITLE
Add basic 2 players management

### PR DESCRIPTION
**Pull request by Chroq**
## Description

In this PR, I'm implementing a basic version of 2 players game.

Trello : [https://trello.com/c/BcbVyuXK/3-05-as-norbert-i-want-some-tips-on-what-to-play-next](url)

### CODE REVIEW

File: board/BottomBoard.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React from "react";
 import {Polygon, Svg} from "react-native-svg";
-import {player1Color, player2Color, strokeColor} from "./theme";
+import {player1Color, player2Color, strokeColor} from "../theme";
```
Comment (floo51) [inline board/BottomBoard.js]: Shouldn't a linter put a space `import {>here<player1Color, player2Color, strokeColor} from "../theme";`?

File: board/index.js [hunk 2]
```diff
@@ -13,15 +13,20 @@ export const Board = (props) => {
     const [isWon, setIsWon] = useState(false);
     const [player, setPlayer] = useState(1);
     const [isLoading, setIsLoading] = useState(false);
+    const [winner, setWinner] = useState(0);
 
     const handleCellOnPress = (cellNumber, player) => {
-        if (isLoading || isWon) {
+        if (isLoading || isWon || grid[cellNumber] !== 0) {
```
Comment (floo51) [inline board/index.js]: Naming this condition `grid[cellNumber] !== 0` would have made it easier to read.

File: home/TrainingButton.js [hunk 3]
```diff
@@ -1,12 +1,15 @@
 import React from "react";
 import {Button} from "react-native";
 
+import {trainingMenu} from "../theme";
+
 export const TrainingButton = ({navigation}) => {
     return <Button
         title="Training mode"
-        color="#999999"
+        color={trainingMenu}
```
Comment (floo51) [inline home/TrainingButton.js]: Naming: I thought you passed a "graphical" Menu component, not a color.

File: home/TwoPlayersLocalButton.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import {Button} from "react-native";
+
+import {sameDevice} from "../theme";
+
+export const TwoPlayersLocalButton = ({navigation}) => {
+    return <Button
+        title="2 players (same device)"
+        color={sameDevice}
```
Comment (floo51) [inline home/TwoPlayersLocalButton.js]: Same as previous comment

File: App.js [hunk 5]
```diff
@@ -1,12 +1,14 @@
 import {createAppContainer} from "react-navigation";
 import {createStackNavigator} from "react-navigation-stack";
 
-import TrainingScreen from "./training";
 import HomeScreen from "./home";
+import TrainingScreen from "./screen/TrainingScreen";
```
Comment (floo51) [inline App.js]: Do you need those 2 almost identical files?
It adds a layer of abstraction making it harder to read what the stack navigator does IMO. I would have loved to see the mode selection directly here, would have helped me understand and avoided me to dig deeper into the project structure.

**Comment by floo51** [comment]
Side note: No, I'm not seeing my grandma this week-end

File: board/index.js [hunk 6]
```diff
@@ -30,16 +35,34 @@ export const Board = (props) => {
             const {isWon} = await fetchIsWon(grid, player);
             setIsWon(isWon);
             setIsLoading(false);
+
+            // @todo: Tricky part to remove
+            if (props.mode === "training") {
+                setWinner(1)
```
Comment (alexisjanvier) [inline board/index.js]: in training mode, player 1 always win ?

File: board/index.js [hunk 7]
```diff
@@ -30,16 +35,34 @@ export const Board = (props) => {
...
+                setWinner(1);
+            }
         });
 
         callApi();
     }, [grid]);
 
     return (
         <Svg width="500" height="500">
+            {!isWon && (
```
Comment (alexisjanvier) [inline board/index.js]: as you have 2 conditions, `!isWon &&` and `isWon &&`, you should use a ternary `isWon ? ... : ...`

Comment (alexisjanvier) [reply board/index.js]: And in general, this logic of displaying the winner should be isolated in a component, for greater readability and to facilitate testing.

File: App.js [hunk 8]
```diff
@@ -1,12 +1,14 @@
...
+import TwoPlayerScreen from "./screen/TwoPlayerScreen";
 
 const MainNavigator = createStackNavigator({
     Home: HomeScreen,
     Training: TrainingScreen,
+    TwoPlayer: TwoPlayerScreen,
```
Comment (alexisjanvier) [inline App.js]: @floo51 is right, it seems you can call a screen for a named route with parameters (https://reactnavigation.org/docs/en/stack-navigator.html#params)

File: board/BottomBoard.js (hunk 1 above)
Comment (Chroq) [reply board/BottomBoard.js]: Fixed in the next PR ! TY (indeed there's space between { and variable name)

File: board/index.js (hunk 2 above)
Comment (Chroq) [reply board/index.js]: If I understand well, it's fixed, no ?

File: home/TrainingButton.js (hunk 3 above)
Comment (Chroq) [reply home/TrainingButton.js]: Renamed ! TY

File: home/TwoPlayersLocalButton.js (hunk 4 above)
Comment (Chroq) [reply home/TwoPlayersLocalButton.js]: Renamed ! TY

File: App.js (hunk 5 above)
Comment (Chroq) [reply App.js]: I have add a todo to refactor ! TY for this feedback

File: board/index.js (hunk 6 above)
Comment (Chroq) [reply board/index.js]: No but I see what you mean ! I fixed that ! TY

File: board/index.js (hunk 7 above)
Comment (Chroq) [reply board/index.js]: I had a todo to make a deep refactoring on logical part ! (Using the reducer and extract the logic into another component) ! TY

File: App.js (hunk 8 above)
Comment (Chroq) [reply App.js]: He's always right ;) I had this link to the todo previously created :+1: TY

File: board/index.js (hunk 2 above)
Comment (floo51) [reply board/index.js]: Nope, I meant something like that:

```
const aStoneIsOnThisSpot = grid[cellNumber] !== 0;
if (isLoading || isWon || aStoneIsOnThisSpot) {
```

**Pull request merged**